gensim/similarities/docsim.py

```diff
@@ -444,8 +447,20 @@ def save(self, fname=None, *args, **kwargs):
         self.close_shard()
         if fname is None:
             fname = self.output_prefix
+
         super(Similarity, self).save(fname, *args, **kwargs)
 
+    @classmethod
+    def load(cls, fname, mmap=None):
+
+        obj = super(Similarity, cls).load(fname, mmap)
+        if isinstance(obj, Similarity):
+            if obj.output_prefix != fname:
```

Comment:
Is this condition needed? IIRC `check_moved` is idempotent if there's no change in the prefix.